src/Files.Uwp/Views/Pages/PropertiesGeneral.xaml.cs

```diff
@@ -115,5 +116,18 @@ private void DiskCleanupButton_Click(object sender, Windows.UI.Xaml.RoutedEventA
                 StorageSenseHelper.OpenStorageSense(drive.Path);
             }
         }
+
+        private async void ChangeApplicationButton_Click(object sender, RoutedEventArgs e)
+        {
+            var filePath = ViewModel.ItemPath + Path.DirectorySeparatorChar + ViewModel.ItemName;
+            await NavigationHelpers.OpenPath(filePath, AppInstance, openViaApplicationPicker: true);
+        }
+
+        private new async void GotFocus(object sender, RoutedEventArgs e)
```

Comment:
An alternative is to use `SHChangeNotifyRegister` in fulltrust process to get notified when a file association changes. But I guess that would be unnecessarily complex.

Reply:
If it's the proper way to do things it might be worth it. I'll look into it.

Reply:
Okay I think you're right about the overly complex issue there, and I think it's above my skill level to implement correctly anyway.